src/runtime/opencl/opencl_common.h

```diff
@@ -128,10 +138,17 @@ class OpenCLWorkspace final : public DeviceAPI {
     }
   }
   // Initialzie the device.
-  void Init();
+  void Init(const std::vector<std::string>& device_types, const std::string& platform_name = "");
+  virtual void Init() {
+    Init({"gpu", "cpu"});
```

Comment:
Only initialize gpu,  to be consistent with existing seeting

Reply:
This initializes only GPU basically, and falls back to CPU only when GPU is not available.  The behavior is consistent with the original one.

Reply:
The main problem is that the current opencl schedules assumes GPU, so they may not work when falling back to CPUs

Reply:
Okay, I'm fine with removing the fallback feature in this PR.